### ISSUE TITLE
Note editor: a word highlighted by spell checker loses bold and italic font style

### ISSUE THREAD

**Issue by michaelkn** [d1vanov/libquentier#41]
![screenshot](https://user-images.githubusercontent.com/25072859/34240949-ee80206a-e622-11e7-828c-bce026ab385d.png)

**Comment by michaelkn** [comment]
The issue is caused by the following CCS style applied to the misspelled word:
```
.misspell {
    font-style: normal;
    font-weight: normal;
    background: url(qrc:/underline.gif) bottom repeat-x;
}

<em class="misspell">mispelled</em>
```
I understand a reason of the font-style override for the `<em>` tag. This tag marks text that has stress emphasis and it is typically displayed in italic type.

I would suggest to replace the `<em>` tag by `<span>` which provides no visual change by itself as mentioned [here](https://www.w3schools.com/tags/tag_span.asp). Then we might get rid of the font properties in CCS:
```
.misspell {
    background: url(qrc:/underline.gif) bottom repeat-x;
}
```

**Comment by d1vanov** [comment]
Thanks for the analysis, the suggestion seems reasonable only need to carefully look into HTML to ENML conversion rules - currently they are set up to ignore `em` tags with `misspell` class.

Also, I recall `em` tags are also used for highlighting of text being searched for, so the highlighted text would also lose bold and italic font style. Probably need to switch to `span` tags there as well.

**Comment by michaelkn** [comment]
I'll try to implement the suggestion and prepare a PR.

**Comment by michaelkn** [comment]
> need to carefully look into HTML to ENML conversion rules - currently they are set up to ignore em tags with misspell class

It seems the tag itself isn't used, only the class:
```
4592 void NoteEditorPrivate::setupSkipRulesForHtmlToEnmlConversion()
...
4618     ENMLConverter::SkipHtmlElementRule spellCheckerHelperSkipRule;              
4619     spellCheckerHelperSkipRule.m_includeElementContents = true;                 
4620     spellCheckerHelperSkipRule.m_attributeValueToSkip = QStringLiteral("misspell");
4621     spellCheckerHelperSkipRule.m_attributeValueCaseSensitivity = Qt::CaseSensitive;
4622     spellCheckerHelperSkipRule.m_attributeValueComparisonRule = ENMLConverter::SkipHtmlElementRule::Contains;
4623     m_skipRulesForHtmlToEnmlConversion << spellCheckerHelperSkipRule;
```

**Comment by michaelkn** [comment]
> `em` tags are also used for highlighting of text being searched for, so the highlighted text would also lose bold and italic font style

The highlighted text doesn't lose italic font style, because there is another css property used: `<em class="hilitorHelper" style="font-style: inherit;`

**Comment by d1vanov** [comment]
> It seems the tag itself isn't used, only the class

Yep, correct. I recall now that I wanted to make these rules account for element name + attribute name + attribute value but then took a shortcut so right now only the attribute value is used in comparison.

> The highlighted text doesn't lose italic font style, because there is another css property used: <em class="hilitorHelper" style="font-style: inherit;

That's good, it's about time CSS has helped something instead of being the source of trouble :blush:

**Comment by d1vanov** [comment]
This one was fixed in commit [0e3b93e](https://github.com/michaelkn/libquentier/commit/0e3b93e144324407ec58e579fb8721801668855a) merged to [development](https://github.com/d1vanov/libquentier/commit/development) branch.
